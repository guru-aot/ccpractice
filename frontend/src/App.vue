<template>
  <div id="app">
    <router-view/>
    <h1>HELLO</h1>
     <h2 style="margin-left: 50px;">User: {{keycloak.tokenParsed.preferred_username}}</h2>
  </div>
</template>

<script>
/* eslint-disable */
import Home from "./components/Home.vue";
import axios from "axios";

export default {
  name: "App",
  props: ["keycloak"],
  components: {
    Home
  },
  data() {
    return {
      user: this.user,
    };
  },
  mounted() {
    axios
      .get("http://localhost:5000/user", {
        headers: {
          Authorization: "Bearer " + this.keycloak.token          
        }
      })
      .then(response => (this.user = response));
  }
};
</script>
