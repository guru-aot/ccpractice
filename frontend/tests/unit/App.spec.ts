import Vue from 'vue';
import { shallowMount } from '@vue/test-utils';
import App from '@/App.vue';
import vuetify from 'vuetify';
import Router from 'vue-router';

Vue.use(vuetify);
Vue.use(Router);
describe('App.vue', () => {
  it('renders without crashing', () => {
    const wrapper = shallowMount(App);
    expect(wrapper.element).toMatchSnapshot();
  });
});
