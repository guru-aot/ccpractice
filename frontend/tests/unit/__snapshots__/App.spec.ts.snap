// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue renders without crashing 1`] = `
<v-app-stub
  id="app"
>
  <header-stub />
   
  <v-content-stub
    tag="main"
  >
    <div
      style="padding: 100px;"
    >
      <router-view-stub
        name="default"
      />
    </div>
  </v-content-stub>
</v-app-stub>
`;
